proposals/data-classes.md

```diff
@@ -195,21 +194,66 @@ var x = new LoginResource {
 Note that `RememberMe` must have an initializer to avoid a warning in the
 object initializer about an unset read-only property.
 
-However, the generated class code would look like:
+**Why have nominal records?**
+
+Nominal records provide two major use cases that are not addressed by positional
+records. The first is adoption convenience. The most common use of a
+record-like type in C# today is a simple class with an object initializer.
+Transforming this into a positional record would require replacing all fields
+with parameters on the primary constructor, replacing all uses of object
+initializers with constructor calls, and explicitly providing default values
+for all parameters which are optional (where this may have been implicitly
+supplied before by simply not putting an initializer on a field or property).
+
+The second, more important use case is API evolution. Positional records make
+it difficult to add or re-order members without producing a breaking change.
+Reordering is essentially impossible for positional records without
+introducing a binary breaking change. Adding new members is difficult and
+requires manual intervention to create constructor overloads with the old
+members which must be carefully maintained. For many uses of records, like
+"options containers" (see
+[Roslyn](http://source.roslyn.io/#Microsoft.CodeAnalysis/Compilation/CompilationOptions.cs,29a3818343614151),
+[Benchmark.NET](https://github.com/dotnet/BenchmarkDotNet/blob/master/src/BenchmarkDotNet/ConsoleArguments/CommandLineOptions.cs)
+for examples), this is a headache. Often data should not have positional
+dependence, and retrofitting independence later is quite difficult.
+
+### Code generation
+
+Aside from places where semantics are underspecified, this section is mostly
+for edification. The implementation of the feature is irrelevant to the
+language design itself. The implementation may change significantly as it is
+actually developed.
+
+That being said, the current proposal is that the generated class code would
+look like:
 
 ```C#
 public class LoginResource : IEquatable<LoginResource>
 {
-    public string <>Backing_Username;
-    public string Username => <>Backing_Username;
-    public string <>Backing_Password;
-    public string Password => <>Backing_Password;
-    public string <>Backing_RememberMe = false;
-    public bool RememberMe => <>Backing_RememberMe;
+    public struct <>Initializer
```

Comment:
my personal preference would be that this just be a well known name that then supports this pattern 'by convention'.  Like so many patterns in the language, we would just be looking for a general accessible/public API-shape, and if found, would light up this functionality.  

So, in this case, having htis be called "Initializer" or "Builder" would be far preferred to some unnameable name like `<>Initializer`.

Reply:
> You realize this is a ridiculously complicated pattern, right?

Not to me :)

> You have to have a public builder.

Not ridiculous.  Not complicated (IMO).

> You have to manually move your initializers from the class to the builder (the compiler can't help with or verify this step).

Neither ridiculous nor complicated :)

> The fields must have the exact same names as the properties. 

Neither ridiculous nor complicated :)

> What if the property has a setter? Is the pattern illegal now (allowing it would be a breaking change)? What message does the compiler produce?

I'm happy to discuss all those potential concerns.  But i would like to start with an open mind that this isn't defacto 'absurd' as you are making it out to be :)

It may be that there is enough cases to help drive the idea that this is better as something truly unnameable.  But i would prefer that be the *result* of a good discussion with enough informign cases, rather than haivng that be the presumed starting point.

Effectively, none of the above seemed at all strange or onerous to have as a defacto pattern.  Heck, it seems exceptionally tame compared to many other patterns we have in the language :)   (Just my 2c on these).

Reply:
I don't underestand what it buys you if it was a pattern? why you'd want to implement it manually? also I think @agocke's point about it being "complicated" is not the pattern itself,  the compiler should be able to extract semantical information from that pattern which could change as the feature involve in the future, defining such pattern can be quite compilicated to be future-proof. on the other hand, as a "lowering strategy" it will come after all the semnatic analysis and also can *change* which has a greater value imo.